public class org.json.JSONException extends java.lang.Exception
{
    private java.lang.Throwable cause;

    public void <init>(java.lang.String)
    {
        org.json.JSONException r0;
        java.lang.String $r1;

        r0 := @this: org.json.JSONException;

        $r1 := @parameter0: java.lang.String;

        specialinvoke r0.<java.lang.Exception: void <init>(java.lang.String)>($r1);

        return;
    }

    public void <init>(java.lang.Throwable)
    {
        java.lang.Throwable $r1;
        org.json.JSONException r0;
        java.lang.String $r2;

        r0 := @this: org.json.JSONException;

        $r1 := @parameter0: java.lang.Throwable;

        $r2 = virtualinvoke $r1.<java.lang.Throwable: java.lang.String getMessage()>();

        specialinvoke r0.<java.lang.Exception: void <init>(java.lang.String)>($r2);

        r0.<org.json.JSONException: java.lang.Throwable cause> = $r1;

        return;
    }

    public java.lang.Throwable getCause()
    {
        java.lang.Throwable r1;
        org.json.JSONException r0;

        r0 := @this: org.json.JSONException;

        r1 = r0.<org.json.JSONException: java.lang.Throwable cause>;

        return r1;
    }
}
